projects/rocfft/clients/tests/CMakeLists.txt

```
@@ -115,7 +115,58 @@ option( BUILD_FFTW "Download and build FFTW" OFF )
 
 # look for installed FFTW if we weren't asked to build it
 if( NOT BUILD_FFTW )
+  # searching FFTW with FindFFTW.cmake that is provided by project
+  # because distro versions of FFTW does not include the cmake specific
+  # helper files
   find_package( FFTW 3.0 MODULE COMPONENTS FLOAT DOUBLE )
+  if( FFTW_FOUND )
+    message(STATUS "FFTW found by FindFFTW.cmake")
+  else()
+    # searching fftw versions build from source code which contain
+    # cmake helper files like FFTW3Config.cmake
+    find_package(FFTW3 CONFIG QUIET )
+    find_package(FFTW3f CONFIG QUIET )
+    if( FFTW3_FOUND AND FFTW3f_FOUND )
+      message(STATUS "FFTW3 found by using the FFTW3Config.cmake")
+      get_target_property(fftw3_libs FFTW3::fftw3 IMPORTED_LOCATION_RELEASE)
+      get_target_property(fftw3f_libs FFTW3::fftw3f IMPORTED_LOCATION_RELEASE)
+
+      # Look for omp (preferred) or thread libraries.  These are not a
+      # hard requirement, but are nice to have to make FFTW run faster.
+      find_library(FFTW_OMP_LIBRARY NAMES fftw3_omp HINTS ${FFTW3_LIBRARY_DIRS})
+      find_library( FFTW_THREADS_LIBRARY fftw3_threads HINTS ${FFTW3_LIBRARY_DIRS} )
+      if( FFTW_OMP_LIBRARY )
+        set( fftw3_mt_libs ${FFTW_OMP_LIBRARY} )
+      elseif( FFTW_THREADS_LIBRARY )
+        set( fftw3_mt_libs ${FFTW_THREADS_LIBRARY} )
+      endif()
+      find_library(FFTWF_OMP_LIBRARY NAMES fftw3f_omp HINTS ${FFTW3f_LIBRARY_DIRS})
+      find_library( FFTWF_THREADS_LIBRARY fftw3f_threads HINTS ${FFTW3f_LIBRARY_DIRS} )
+      if( FFTWF_OMP_LIBRARY )
+        if ( fftw3_mt_libs )
+          list( APPEND fftw3_mt_libs ${FFTWF_OMP_LIBRARY} )
+          set( FFTW_MULTITHREAD TRUE )
+        endif()
+      elseif( FFTWF_THREADS_LIBRARY )
+        if ( fftw3_mt_libs )
+          list( APPEND fftw3_mt_libs ${FFTWF_THREADS_LIBRARY} )
+          set( FFTW_MULTITHREAD TRUE )
+        endif()
+      endif()
```

What about the following:
```suggestion
      find_library(FFTW_OMP_LIBRARY NAMES fftw3_omp HINTS ${FFTW3_LIBRARY_DIRS})
      find_library(FFTWF_OMP_LIBRARY NAMES fftw3f_omp HINTS ${FFTW3f_LIBRARY_DIRS})
      if ( FFTW_OMP_LIBRARY AND FFTWF_OMP_LIBRARY )
          list( APPEND FFTW_LIBRARIES ${FFTW_OMP_LIBRARY} )
          list( APPEND FFTW_LIBRARIES ${FFTWF_OMP_LIBRARY} )
          set( FFTW_MULTITHREAD TRUE )
      else()
          find_library( FFTW_THREADS_LIBRARY fftw3_threads HINTS ${FFTW3_LIBRARY_DIRS} )
          find_library( FFTWF_THREADS_LIBRARY fftw3f_threads HINTS ${FFTW3f_LIBRARY_DIRS} )
          if ( FFTW_THREADS_LIBRARY AND FFTWF_THREADS_LIBRARY )
              list( APPEND FFTW_LIBRARIES ${FFTW_THREADS_LIBRARY } )
              list( APPEND FFTW_LIBRARIES ${FFTWF_THREADS_LIBRARY } )
              set( FFTW_MULTITHREAD TRUE )
          endif()
      endif()
```
This makes it a bit easier to see if both libs are found

---

In theory one could build fftw with omp and fftwf with threads or viceversa and then this would not work :-)